accounts-db/src/accounts_index/bucket_map_holder.rs

```diff
@@ -82,6 +88,38 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> BucketMapHolder<T, U>
         self.disk.is_some()
     }
 
+    /// Check if flushing to disk should occur based on entry count threshold per bin
+    pub fn should_flush_to_disk(&self, entries_in_bin: usize, high_water_mark_ratio: f64) -> bool {
```

Comment:
I'm still not a fan of putting the high water mark here. IMO we can compute the max per bin, the high water mark, and the low water mark all at the same time, and just once. Then keep them all as a field in the BucketMapHolder.

We can pass the *number* of entries for the hwm and lwm to their respective fns, and bypass any floating point math. Should still be as testable as before. And then we end up with all threshold values/constants together. (Currently the three values are all in different locations, plus the two ratios are literals without comments.)